ckanapi/cli/main.py

```diff
@@ -130,6 +134,27 @@ def main(running_with_paster=False):
             )
     else:
         ckan = LocalCKAN(username=arguments['--ckan-user'])
+        # log execution of LocalCKAN commands
+        from ckan.plugins.toolkit import config, asbool
+        if asbool(config.get('ckanapi.log_local')) and len(sys.argv) > 1:
+            cmd = ['who', 'am', 'i']
+            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            out, err = proc.communicate()
+            if not out or err:
+                # fallback to whoami if `who am i` is empty or errored
+                cmd = ['whoami']
+                proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE)
+                out, err = proc.communicate()
+            if not out or err:
+                # cannot find user
+                out = '<unknown user>'
+            else:
+                # remove line breaks from whoami's
+                out = out.replace('\n', '').replace('\r', '')
```

Comment:
`who am i` returns a line of text like
```
myusername      pts/14       2024-03-13 12:40
```
so you'd need to split at the first space

Reply:
Added `split()[0]` for the output after replacing line breaks